### PR TITLE
feat: add support for arrays as children

### LINKED ISSUE #3: Support Array of Attributes
@texastoland showed me your readme and we discussed how it would be cool to support an array of attributes.

In the example in the read me:

```jsx
import jsxobj from 'jsxobj';

// example of an import'd plugin
const CustomPlugin = config => ({
	...config,
	name: 'custom-plugin'
});

console.log(
	<webpack target="web" watch>
		<entry path="src/index.js" />
		<resolve>
			<alias from="react" to="preact-compat" />
			<alias from="react-dom" to="preact-compat" />
		</resolve>
		<plugins>
			<uglify-js opts={{
				compression: true,
				mangle: false
			}} />
			<CustomPlugin foo="bar" />
		</plugins>
	</webpack>
);
```

The `resolve` attribute should turn into an array of aliases:

```jsx
<resolve>
  <alias from="react" to="preact-compat" />
  <alias from="react-dom" to="preact-compat" />
</resolve>
```
into:

```json
"resolve": {
    "alias": [
      {
        "from": "react",
        "to": "preact-compat"
      },
      {
        "from": "react-dom",
        "to": "preact-compat"
      }
    ]
  },
```

We can support this by adding a check for if the `obj[name]` is already defined. 

```js
/** @jsx h */
function h(name, attrs, ...children) {
    let obj = { name, ...attrs };
    if (typeof name==='function') obj = name(obj);
    [].concat(...children).forEach( child => {
        let { name } = child;
        if (name) {
          if(obj[name]){
            obj[name] = [obj[name], child].flat()
          } else {
            obj[name] = child;
          }        
            delete child.name;
        }
        else obj.value = child;
    });
    return obj;
}
```

Heres a [js fiddle example](https://jsfiddle.net/theianjones/yos4g6ra/5/)

### PR DESCRIPTION
**Pull request by theianjones**
resolves #3 

party

### CODE REVIEW

**Comment by texastoland** [comment]
@theianjones I liked your implementation too 👏🏼 Look for my review inline.

File: src/index.js [hunk 1]
```diff
@@ -9,10 +9,14 @@ export default function h(name, attrs, ...children) {
...
+      if(obj[name]){
+        obj[name] = [obj[name], child].flat()
+      } else {
+        obj[name] = child;
+      }
+
 			delete child.name;
-		}
-		else {
+		}		else {
```
Comment (texastoland) [inline src/index.js]: Do you not have the Prettier extension installed? Make sure to add an `.editorconfig` to keep tabs (you mixed them throughout).

File: src/index.js [hunk 2]
```diff
@@ -9,10 +9,14 @@ export default function h(name, attrs, ...children) {
 		// collapse it into the current object for MAGIC
 		let { name } = child;
 		if (name) {
-			obj[name] = child;
+      if(obj[name]){
+        obj[name] = [obj[name], child].flat()
+      } else {
+        obj[name] = child;
+      }
```
Comment (texastoland) [inline src/index.js]: I might not have thought of `flat` 🤩

```suggestion
		obj[name] = obj[name] ? [obj[name], child].flat() : child
```

File: test/index.js [hunk 3]
```diff
@@ -1,33 +1,50 @@
...
+        <resolve>
+                <alias from="react-dom" to="preact-compat" />
+        </resolve>
+        <plugins>
+                <uglify-js opts={{
+                        compression: true,
+                        mangle: false,
+                }} />
+                <CustomPlugin foo="bar" />
+        </plugins>
```
Comment (texastoland) [inline test/index.js]: Unintentional whitespace changes hurt the diff.

File: test/arrayFixture.json [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    "alias": [
+      {
+        "from": "react",
+        "to": "preact-compat"
+      },
+      {
+        "from": "react-dom",
+        "to": "preact-compat"
+      }
+    ]
```
Comment (texastoland) [inline test/arrayFixture.json:18]: Readme needs updated too 👏🏼

**Comment by theianjones** [comment]
@texastoland I dont see a review

**Review by texastoland — CHANGES_REQUESTED** [review]

**Pull request merged**